packages/calcite-components/src/components/action-menu/action-menu.tsx

```diff
@@ -319,6 +319,8 @@ export class ActionMenu extends LitElement implements LoadableComponent {
       "keydown",
       this.menuButtonKeyDown,
     ) /* TODO: [MIGRATION] If possible, refactor to use on* JSX prop or this.listen()/this.listenOn() utils - they clean up event listeners automatically, thus prevent memory leaks */;
+
+    this.menuButtonEl = null;
```

Comment:
Not clearing this reference would cause `connectMenuButtonEl` to bail since the references would [remain the same](https://github.com/Esri/calcite-design-system/blob/jcfranco/clear-action-menu-el-on-disconnect/packages/calcite-components/src/components/action-menu/action-menu.tsx#L266-L268).